activerecord/lib/active_record/relation.rb

```diff
@@ -572,9 +573,20 @@ def touch_all(*names, time: nil)
     #
     # ==== Examples
     #
-    #   Person.where(age: 0..18).destroy_all
-    def destroy_all
-      records.each(&:destroy).tap { reset }
+    #   Person.where(age: 0..18).destroy_all(batch_size: 3000)
+    def destroy_all(batch_options = {})
+      if ActiveRecord::Base.destroy_all_in_batches
+        in_batches(**batch_options).each_record(&:destroy).tap { reset }
```

Comment:
@rafaelfranca @albertoalmagro I was wondering about making the API more explicit here by specifying the possible options that in_batches is expecting? similar to https://github.com/rails/rails/blob/83217025a171593547d1268651b446d3533e2019/activerecord/lib/active_record/relation/batches.rb#L128
```suggestion
    def destroy_all(start: nil, finish: nil, batch_size: 1000, error_on_ignore: nil, order: :asc)
      if ActiveRecord::Base.destroy_all_in_batches
        in_batches(of: batch_size, start: start, finish: finish, load: true, error_on_ignore: error_on_ignore, order: order).each_record(&:destroy).tap { reset }
```

Also document the possible options that now destroy_all is accepting wdyt?

Reply:
Yes, please update the method signature and add the comment for options.

Reply:
@guilleiguaran done in 744c16b1ac0832303b2c0da854fc508daedb3cb1